lib/light-service/organizer/iterate.rb

```diff
@@ -1,22 +1,82 @@
+# require 'light-service/action.rb'
+
 module LightService
   module Organizer
-    class Iterate
-      extend ScopedReducable
+    class Iterator
+      using ::LightService::Refinements::Array
+
+      include ScopedReducable
+
+      def initialize(collection_key, actions)
+        @collection_key = collection_key
+        @actions = actions
+        @iterations = nil
+        @current_iteration = nil
+      end
+
+      def rollback(ctx)
+        reversable_iterations.reverse.each_with_index do |single_iteration, idx|
+          ctx[single_iteration.item_key] = single_iteration.item
+
+          unless single_iteration.actions.include?(ctx.current_action)
+            ctx.current_action = single_iteration.actions.last
+          end
+
+          ctx.current_action = single_iteration.actions.last unless idx.zero?
```

Comment:
`idx.zero?` alias "Am I on the first interrupted `SingleIteration`?"

Reply:
Going to give an exmplanation of `single_iteration.actions.include?(ctx.current_action)` and `idx.zero?`

Having this iteration where numbers are `item` and letters are `Action`s

```
1 [ A B C ]
2 [ A B C ]
3 [ A B C ]
```

failing with rollback on `2B`, thus having `B` as `current_action` in `context`.

Start the rollback for `item` 2:

* "Was the whole process interrupted while inside the iterator?" yes; thus preserve the `current_action`
* "Am I on the first interrupted SingleIteration?" yes I am; thus preserve the `current_action`

The `SubOrganizer` will rollback `2B` then `2A`.

Next cycle will be on item `1`:

* "Was the whole process interrupted while inside the iterator?" yes; thus preserve the `current_action`
* * "Am I on the first interrupted SingleIteration?" no, I am on the second (reversed order), so reset the `current_action` to the first (reversed order) `Action` of this `SingleIteration`

The `SubOrganizer` will rollback `1C` then `1B` then `1A`.

Reply:
Should I switch back to the original `current_action`? 🤷

Reply:
Code changed, but the reasoning is still useful. I've added some comment inline to clarify most obsure logic

Reply:
Nevermind; the code has comments. Let's cleanup the PR